src/Domain/ElementLogicalCoordinates.hpp

```diff
@@ -0,0 +1,47 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <cstddef>
+#include <unordered_map>
+#include <vector>
+
+#include "DataStructures/Tensor/TypeAliases.hpp"
+
+class DataVector;
+template <size_t Dim>
+struct BlockLogicalCoordHolder;
+template <size_t VolumeDim>
+class ElementId;
+
+/// \ingroup ComputationalDomainGroup
+///
+/// Holds element logical coordinates of an arbitrary set of points on
+/// a single `Element`.  The arbitrary set of points is assumed to be
+/// a subset of a larger set of points spanning multiple `Element`s,
+/// and this class holds `offsets` that index into that larger set of
+/// points.
+///
+/// \details `offsets.size()` is the same as
+/// `element_logical_coords.number_of_grid_points()`
+template <size_t Dim>
+struct ElementLogicalCoordHolder {
+  tnsr::I<DataVector, Dim, Frame::Logical> element_logical_coords;
+  std::vector<size_t> offsets;
+};
+
+/// \ingroup ComputationalDomainGroup
+///
+/// Given a set of points in block logical coordinates, and their
+/// block_ids, determines which elements in a list of ElementIds
+/// contains each point, and determines the element logical
+/// coordinates of each point.
+///
+/// \details Returns a std::ordered_map from ElementIds to
+/// ElementLogicalCoordHolders.
+template <size_t Dim>
```

Comment:
Also should note that it is fine that a subset of the input coordinates will be found in the elements